Everest tests overbooks CPU

When running
`just everest-tests`

it runs the test with 4 pytest runners, suitable for hardware with 4 cpus/cores. However, when monitoring the load when running this, the load approaches 20. This means the CPU is overbooked, and total runtime is longer than necessary, and it will also cause flakyness due to things timing out.

Example:
https://github.com/equinor/ert/actions/runs/17034411655/job/48285156523

```
--------------------------- Captured stdout teardown ---------------------------
System load after test failure (1/5/15min): 17.13, 8.47, 3.31, cpu_count=4
----------------------------- Captured stdout call -----------------------------
```


```
FAILED tests/everest/test_everest_client.py::test_that_multiple_everest_clients_can_connect_to_server@math_func/config_minimal.yml - AssertionError: assert [] == ['EverestStat...usEvent', ...]
  
  Right contains 13 more items, first extra item: 'EverestStatusEvent'
  
  Full diff:
  + []
  - [
  -     'EverestStatusEvent',
  -     'EverestStatusEvent',
  -     'EverestBatchResultEvent',
  -     'EverestBatchResultEvent',
  -     'EverestStatusEvent',
  -     'EverestStatusEvent',
  -     'EverestBatchResultEvent',
  -     'EverestStatusEvent',
  -     'EverestStatusEvent',
  -     'EverestBatchResultEvent',
  -     'EverestStatusEvent',
  -     'EverestStatusEvent',
  -     'EverestBatchResultEvent',
  - ]
== 1 failed, 534 passed, 14 skipped, 2 warnings, 1 rerun in 414.37s (0:06:54) ==
error: Recipe `everest-tests` failed on line 35 with exit code 1
```

A recent failure in https://github.com/equinor/komodo-releases/actions/runs/17085683947/job/48451551814

`FAILED tests/everest/test_everserver.py::test_status_too_few_realizations_succeeded@starts_everest - assert 'Optimization failed: not enough successful realizations to proceed.' in 'Traceback (most recent call last):\n  File "/prog/res/komodo/bleeding-20250820-0306-py311-rhel8-numpy2/root/lib64/pyt...eptions\n    raise mapped_exc(message) from exc\nhttpx.ConnectTimeout: _ssl.c:999: The handshake operation timed out\n'`

is possible to reproduce with:
`pytest tests/everest/test_everserver.py -k too_few --count 100 -n 16
`
and is seemingly solved by setting max_running to 1

--- a/tests/everest/conftest.py
+++ b/tests/everest/conftest.py
@@ -17,6 +17,7 @@ import yaml
 
 import ert
 import everest
+from ert.config.queue_config import LocalQueueOptions
 from ert.ensemble_evaluator import EvaluatorServerConfig
 from ert.run_models import StatusEvents
 from ert.run_models.event import status_event_from_json, status_event_to_json
@@ -200,6 +201,7 @@ def cached_example(pytestconfig):
 
             shutil.copytree(config_path.parent, my_tmpdir / "everest")
             config = EverestConfig.load_file(my_tmpdir / "everest" / config_file)
+            config.simulator.queue_system = LocalQueueOptions(max_running=2)
             status_queue: queue.SimpleQueue[StatusEvents] = queue.SimpleQueue()
             run_model = EverestRunModel.create(config, status_queue=status_queue)
             evaluator_server_config = EvaluatorServerConfig()

--- a/tests/everest/functional/test_main_everest_entry.py
+++ b/tests/everest/functional/test_main_everest_entry.py
@@ -51,7 +51,7 @@ def test_everest_entry_run(cached_example):
 
     # Ensure no interference with plugins which may set queue system
     config_content = yaml.safe_load(Path(config_file).read_text(encoding="utf-8"))
-    config_content["simulator"] = {"queue_system": {"name": "local"}}
+    config_content["simulator"] = {"queue_system": {"name": "local", "max_running": 2}}
     Path(config_file).write_text(
         yaml.dump(config_content, default_flow_style=False), encoding="utf-8"
     )

--- a/tests/everest/test_everest_client.py
+++ b/tests/everest/test_everest_client.py
@@ -158,7 +158,7 @@ def test_that_multiple_everest_clients_can_connect_to_server(
 
     config_path = Path(path) / config_file
     config_content = yaml.safe_load(config_path.read_text(encoding="utf-8"))
-    config_content["simulator"] = {"queue_system": {"name": "local"}}
+    config_content["simulator"] = {"queue_system": {"name": "local", "max_running": 2}}
     config_path.write_text(
         yaml.dump(config_content, default_flow_style=False), encoding="utf-8"
     )

--- a/tests/everest/test_everserver.py
+++ b/tests/everest/test_everserver.py
@@ -178,7 +178,7 @@ async def test_status_max_batch_num(copy_math_func_test_data_to_tmp):
     config_dict = {
         **config.model_dump(exclude_none=True),
         "optimization": {"algorithm": "optpp_q_newton", "max_batch_num": 1},
-        "simulator": {"queue_system": {"name": "local"}},
+        "simulator": {"queue_system": {"name": "local", "max_running": 2}},
     }
     config = EverestConfig.model_validate(config_dict)
 
@@ -207,7 +207,7 @@ async def test_status_too_few_realizations_succeeded(copy_math_func_test_data_to
     config_dict = {
         **config.model_dump(exclude_none=True),
         "optimization": {"algorithm": "optpp_q_newton", "max_batch_num": 1},
-        "simulator": {"queue_system": {"name": "local"}},
+        "simulator": {"queue_system": {"name": "local", "max_running": 2}},
         "model": {"realizations": [0, 1]},
     }
     config_dict["install_jobs"].append(
@@ -236,7 +236,7 @@ async def test_status_all_realizations_failed(copy_math_func_test_data_to_tmp):
     config_dict = {
         **config.model_dump(exclude_none=True),
         "optimization": {"algorithm": "optpp_q_newton", "max_batch_num": 1},
-        "simulator": {"queue_system": {"name": "local"}},
+        "simulator": {"queue_system": {"name": "local", "max_running": 2}},
     }
     config_dict["install_jobs"].append({"name": "fail", "executable": which("false")})
     config_dict["forward_model"].append("fail")
@@ -258,7 +258,10 @@ async def test_status_all_realizations_failed(copy_math_func_test_data_to_tmp):
 @pytest.mark.timeout(240)
 @patch("sys.argv", ["name", "--output-dir", "everest_output"])
 async def test_status_contains_max_runtime_failure(change_to_tmpdir, min_config):
-    min_config["simulator"] = {"queue_system": {"name": "local"}, "max_runtime": 1}
+    min_config["simulator"] = {
+        "queue_system": {"name": "local", "max_running": 2},
+        "max_runtime": 1,
+    }
     min_config["forward_model"] = ["sleep 5"]
     min_config["install_jobs"] = [{"name": "sleep", "executable": which("sleep")}]
 

--- a/tests/everest/test_math_func.py
+++ b/tests/everest/test_math_func.py
@@ -140,6 +140,7 @@ def test_math_func_auto_scaled_controls(copy_math_func_test_data_to_tmp):
                 "upper_bound": 0.5,
             }
         ],
+        "simulator": {"queue_system": {"name": "local", "max_running": 2}},
     }
     config = EverestConfig.model_validate(config_dict)
 
@@ -168,11 +169,12 @@ def test_math_func_auto_scaled_objectives(copy_math_func_test_data_to_tmp):
     config = EverestConfig.load_file("config_multiobj.yml")
     config_dict = config.model_dump(exclude_none=True)
 
+    config_dict["simulator"] = {"queue_system": {"name": "local", "max_running": 2}}
+
     # Normalize only distance_p:
     config_dict["objective_functions"][0]["auto_scale"] = True
     config_dict["objective_functions"][0]["scale"] = 1.0
     config_dict["optimization"]["max_batch_num"] = 1
-
     config = EverestConfig.model_validate(config_dict)
     run_model = EverestRunModel.create(config)
     evaluator_server_config = EvaluatorServerConfig()
@@ -191,6 +193,8 @@ def test_math_func_auto_scaled_objectives(copy_math_func_test_data_to_tmp):
 def test_math_func_auto_scaled_constraints(copy_math_func_test_data_to_tmp):
     config = EverestConfig.load_file("config_advanced.yml")
     config_dict = config.model_dump(exclude_none=True)
+
+    config_dict["simulator"] = {"queue_system": {"name": "local", "max_running": 2}}
 
     # control number of batches, no need for full convergence:
     config_dict["optimization"]["convergence_tolerance"] = 1e-10
@@ -239,6 +243,8 @@ def test_that_math_func_violating_output_constraints_has_no_result(
     config = EverestConfig.load_file("config_advanced.yml")
     config_dict = config.model_dump(exclude_none=True)
 
+    config_dict["simulator"] = {"queue_system": {"name": "local", "max_running": 2}}
+
     # The first batch violates the output constraint:
     config_dict["optimization"]["max_batch_num"] = 1
     config_dict["controls"][0]["initial_guess"] = 0.05
@@ -257,6 +263,8 @@ def test_that_math_func_violating_output_constraints_has_a_result(
 ):
     config = EverestConfig.load_file("config_advanced.yml")
     config_dict = config.model_dump(exclude_none=True)
+
+    config_dict["simulator"] = {"queue_system": {"name": "local", "max_running": 2}}
 
     # The second batch does not violate the output constraint:
     config_dict["optimization"]["max_batch_num"] = 2
@@ -277,6 +285,8 @@ def test_that_math_func_violating_input_constraints_has_no_result(
     config = EverestConfig.load_file("config_advanced.yml")
     config_dict = config.model_dump(exclude_none=True)
 
+    config_dict["simulator"] = {"queue_system": {"name": "local", "max_running": 2}}
+
     # The first batch violates the input constraint:
     config_dict["optimization"]["max_batch_num"] = 1
     config_dict["controls"][0]["initial_guess"] = 0.5
@@ -295,6 +305,8 @@ def test_that_math_func_violating_input_constraints_has_a_result(
 ):
     config = EverestConfig.load_file("config_advanced.yml")
     config_dict = config.model_dump(exclude_none=True)
+
+    config_dict["simulator"] = {"queue_system": {"name": "local", "max_running": 2}}
 
     # The second batch does not violate the input constraint:
     config_dict["optimization"]["max_batch_num"] = 2

--- a/tests/everest/test_objective_type.py
+++ b/tests/everest/test_objective_type.py
@@ -27,6 +27,7 @@ def test_objective_type(copy_math_func_test_data_to_tmp):
                 "--target 0.5 0.5 0.5 --out stddev"
             ),
         ],
+        "simulator": {"queue_system": {"name": "local", "max_running": 2}},
     }
     config = EverestConfig.model_validate(config_dict)
     # Act

